src/controllers/RaceController.js

```diff
@@ -0,0 +1,51 @@
+import { Random } from "@woowacourse/mission-utils";
+
+class RaceController {
+
+  createPlayersPositions(playerNumber) {
+    const positions = Array(playerNumber).fill(0);
+    return positions;
+  }
+
+  updatePlayersPositions(positions, i) {
+    const num = Random.pickNumberInRange(0, 9);
+
+    if (num >= 4) {
+      positions[i] += 1;
+    }
+  }
+
+  movePlayersRandomly(peopleList) {
+
+    const updatedList = [...peopleList];
+
+    for (let i = 0; i < peopleList.length; i++) {
+      this.updatePlayersPositions(updatedList, i);
+    }
+
+    return updatedList;
+  }
+
+  getMaxValue(positions) {
+    return Math.max(...positions);
+  }
+
+  addWinnerIndex(value, maxValue, index, winnersIndex) {
+    if (value === maxValue) {
+      winnersIndex.push(index);
+    }
+  }
+
+  getRaceWinnersIndex(positions) {
+    const maxValue = this.getMaxValue(positions);
+    const winnersIndex = [];
+
+    positions.forEach((value, index) => {
+      this.addWinnerIndex(value, maxValue, index, winnersIndex);
+    });
+    return winnersIndex;
+
+  }
```

Comment:
```suggestion
const getWinnerIndex = (value, maxValue, index) => {
  if (value === maxValue) {
    return index;
  }
  return null;
};

  getRaceWinnersIndexes(positions) {
    const maxValue = this.getMaxValue(positions);
    return positions
    .map((value, index) => getWinnerIndex(value, maxValue, index))
    .filter((winnerIndex) => winnerIndex !== null);
  }
```

현재 `addWinnerIndex` 함수는 외부의 `winnersIndex` 배열을 직접 수정하여 사이드 이펙트를 유발하고 있습니다. 이는 함수의 재사용성과 디버깅을 어렵게 만들 수 있는 비순수 함수입니다.
동일한 입력에 대해 항상 동일한 출력을 반환하는 순수 함수로 작성해보시는 것은 어떻게 생각하시나요?

Reply:
> 현재 `addWinnerIndex` 함수는 외부의 `winnersIndex` 배열을 직접 수정하여 사이드 이펙트를 유발하고 있습니다. 이는 함수의 재사용성과 디버깅을 어렵게 만들 수 있는 비순수 함수입니다. 동일한 입력에 대해 항상 동일한 출력을 반환하는 순수 함수로 작성해보시는 것은 어떻게 생각하시나요?

순수함수에 대한 이해가 없어서 해당 부분을 이번 리뷰를 통해 배우게되는군요 피드백 감사합니다.